### DISCUSSION TITLE
✅ YubiKey OTP login support

### DISCUSSION THREAD

**Discussion by diegodelima** [isontheline/pro.webssh.net#1052] [Ideas]
Webssh asks me correctly for my OTP code during login, but doesn't have the option to use the code from my YubyKey via NFC.

**Comment by isontheline** [comment]
Hello @diegodelima 👋

Thanks for your feedback 🙏

YubiKeys are not supported at this time.
I can however try to work on them.

Which Yubico key are you using?

**Comment by diegodelima** [comment]
Hi!

It's YubiKey 5C NFC, just to store the OTP 6 digit code.

**Reply by isontheline** [reply]
Thank you @diegodelima 👍
I'm purchasing one in order to make so work in a few days
I've just created an issue to track progress : https://github.com/isontheline/pro.webssh.net/issues/1053

**Reply by isontheline** [reply]
Just bought but ETA of delivery jump to March 😓

**Comment by diegodelima** [comment]
Hi Arnaud,

Looks like Yubikey linux PAM module (pam_yubico.so) behaves very much like the pam_google_authenticator.so with the keyboard-interactive sshd Auth Method, except for the message:

***@***.*** % ssh ***@***.***
***@***.***) Verification code: (this is google auth)
***@***.***) YubiKey for `me’:  (this is yubikey)

WebSSH already shows the TEXT prompt for Yubikey correctly on screen (if I copy and paste the code from the computer it works). What’s missing is a NFC button there, one that calls the iOS layer to ask for it (I guess any NFC key would do the trick for debugging purposes).

I can beta test it with my Yubikey if you want.

**Comment by isontheline** [comment]
Hi @diegodelima Diego 👋

Yes I will add the Yubikit inside WebSSH in order to let you use your Yubikey : https://github.com/Yubico/yubikit-ios
I will try to make a beta release in a few days and I will let you test it

Thank you so much 🙏

**Reply by isontheline** [reply]
Hi @diegodelima 👋

I've worked on it last night and pushed some tests on TestFlight : 
https://testflight.apple.com/join/QSrBK59z

Could you give it a try?

**Comment by diegodelima** [comment]
Hi!

I downloaded the TestFlight, but didn’t seem to work. Still asks for a keyboard input with no "NFC" button.

**Reply by isontheline** [reply]
Excuse me @diegodelima 

It's 25.3.1276 : 
https://testflight.apple.com/join/QSrBK59z

**Comment by diegodelima** [comment]
Working perfectly!!!

**Reply by isontheline** [reply]
Glad to hear that 😅 🙏
